Assets/HoloToolkit/Input/Scripts/Interactions/HandDraggable.cs

```diff
@@ -205,15 +218,56 @@ private void UpdateDragging()
                 draggingRotation = Quaternion.LookRotation(objForward, objUp);
             }
 
+            Vector3 newPosition = draggingPosition + mainCamera.transform.TransformDirection(objRefGrabPoint);
+            Vector3 lastPosition = HostTransform.position;
+
             // Apply Final Position
-            HostTransform.position = draggingPosition + mainCamera.transform.TransformDirection(objRefGrabPoint);
+            HostTransform.position = newPosition;
             HostTransform.rotation = draggingRotation;
 
             if (IsKeepUpright)
             {
                 Quaternion upRotation = Quaternion.FromToRotation(HostTransform.up, Vector3.up);
                 HostTransform.rotation = upRotation * HostTransform.rotation;
             }
+
+            // Check for a possible collision
+            if (WatchForCollision && CollideTransform != null)
+            {
+                // Get bounds for CollideTransform
+                Bounds bounds = new Bounds(CollideTransform.position, Vector3.zero);
+                Renderer[] renderers = CollideTransform.gameObject.GetComponentsInChildren<Renderer>();
+                foreach (Renderer render in renderers)
+                {
+                    bounds.Encapsulate(render.bounds);
+                }
+
+                Vector3 movement = newPosition - lastPosition;
+                float distance = movement.magnitude;
+                movement.Normalize();
+
+                // Check for collision
+                // Only via Physics.CheckBox isn't enough, because the user could be moving the object so fast
+                // that the object is moving through the possible colliding object and we wouldn't hit it.
+                if (Physics.CheckBox(bounds.center, bounds.extents, Quaternion.identity, CollisionMask) ||
+                    Physics.Raycast(lastPosition, movement, distance, CollisionMask))
+                {
+                    // If we collided with something, set back the position to the last one.
+                    HostTransform.position = lastPosition;
+
+                    // Save source temporary so we can reset and set the current position
```

Comment:
This will fire DragStopped and DragStarted every time there is a collision. There is probably a better way to update the start position that has fewer side-effects.